### PR TITLE
BUIP018 Seeding

### PR DESCRIPTION
**Pull request by bitcartel**
Bitnodes API seeding and user-configurable DNS seeds.
https://bitco.in/forum/threads/buip018-seeding.1065/

### CODE REVIEW

File: src/net.cpp [hunk 1]
```diff
@@ -1391,6 +1398,42 @@ void MapPort(bool)
 }
 #endif
 
+// BITCOINUNLIMITED START
+void ThreadBitnodesAddressSeed()
+{
+    // Get nodes from websites offering Bitnodes API
+    if ((addrman.size() > 0) &&
+        (!GetBoolArg("-forcebitnodes", DEFAULT_FORCEBITNODES))) {
+        MilliSleep(11 * 1000);
```
Comment (gandrewstone) [inline src/net.cpp:1407]: why 11 seconds?

File: src/bitnodes.cpp [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+/**
+ * Pass in an empty string vector to be populated with ip:port strings
+ */
+bool GetLeaderboardFromBitnodes(vector<string>& vIPs)
+{
+    // Bitnodes connection parameters
+    string url_host = "bitnodes.21.co";
+    string url_port = "443";
+    string url_path = "/api/v1/nodes/leaderboard/?limit=100";
+    string cert_hostname = "dazzlepod.com";
```
Comment (gandrewstone) [inline src/bitnodes.cpp:171]: what is dazzlepod?

Comment (bitcartel) [inline src/bitnodes.cpp:171]: It's the hostname on the certificate.  I spoke with Bitnodes creator @ayeowch about this a while ago:

> ...that is how the PTR record has always been configured and will remain the same as long as the project is up.  So yes, you can use dazzlepod.com for hostname check.

File: src/net.cpp (hunk 1 above)
Comment (bitcartel) [inline src/net.cpp:1407]: It's a magic number.  For consistency I used the same value as found in ThreadDNSAddressSeed().

https://github.com/BitcoinUnlimited/BitcoinUnlimited/blob/0.12bu/src/net.cpp#L1400

**Comment by gandrewstone** [comment]
Awesome, thx for the submission!

**Comment by rebroad** [comment]
I like this feature - I'm trying to get it to work on Bitcoin Core but it doesn't compile easily for some reason :-s

**Comment by ptschip** [comment]
it might be your boost version...i think this only works on boost 1.59
or above

**Pull request merged**